src/wallet_client.ts

```diff
@@ -174,32 +174,56 @@ export class WalletClient {
   async transferSuiMnemonic(
     amount: number,
     suiAccount: Ed25519Keypair,
-    receiverAddress: SuiAddress
+    receiverAddress: SuiAddress,
+    typeArg: string = SUI_TYPE_ARG
   ) {
     const keypair = suiAccount;
     const senderAddress = keypair.getPublicKey().toSuiAddress();
-    const coinsNeeded =
+    if(typeArg === SUI_TYPE_ARG){
```

Comment:
nit pick on the coding style: (note on the spacing)
if (typeArg === SUI_TYPE_ARG) {

} else {

}

Reply:
Ok will remember that in future